generator/src/main/java/org/corfudb/generator/operations/CheckpointOperation.java

```diff
@@ -0,0 +1,35 @@
+package org.corfudb.generator.operations;
+
+import org.corfudb.generator.State;
+import org.corfudb.runtime.MultiCheckpointWriter;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Created by maithem on 7/14/17.
+ */
+@Slf4j
+public class CheckpointOperation extends Operation {
+
+    public CheckpointOperation(State state) {
+        super(state);
+    }
+
+    @Override
+    public void execute() {
+        try {
+            MultiCheckpointWriter mcw = new MultiCheckpointWriter();
+            mcw.addAllMaps(state.getMaps());
+            long trimAddress = mcw.appendCheckpoints(state.getRuntime(), "Maithem");
+            Thread.sleep(1000 * 30 * 1);
+            state.getRuntime().getAddressSpaceView().prefixTrim(trimAddress - 1);
+            state.setTrimMark(trimAddress);
+            state.getRuntime().getAddressSpaceView().gc();
+            state.getRuntime().getAddressSpaceView().invalidateClientCache();
+            state.getRuntime().getAddressSpaceView().invalidateServerCaches();
+            System.out.println("CheckpointOperation Completed");
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)